### PR TITLE
Enable AOT analysis and fix warnings

### PR DESCRIPTION
**Pull request by JamesNK**

### CODE REVIEW

File: build/dependencies.props [hunk 1]
```diff
@@ -15,7 +15,8 @@
     <MicrosoftCodeAnalysisNetAnalyzersPackageVersion>7.0.0-preview1.22464.1</MicrosoftCodeAnalysisNetAnalyzersPackageVersion>
     <MicrosoftCrankEventSourcesPackageVersion>0.2.0-alpha.21255.1</MicrosoftCrankEventSourcesPackageVersion>
     <MicrosoftExtensionsLoggingTestingPackageVersion>2.1.1</MicrosoftExtensionsLoggingTestingPackageVersion>
-    <MicrosoftExtensionsPackageVersion>3.0.3</MicrosoftExtensionsPackageVersion>
+    <MicrosoftExtensionsPackageVersion>7.0.0-rc.2.22472.3</MicrosoftExtensionsPackageVersion>
+    <MicrosoftExtensions3PackageVersion>3.0.3</MicrosoftExtensions3PackageVersion>
```
Comment (BrennanConroy) [inline build/dependencies.props:19]: Can you at least use 3.1.x?

File: testassets/InteropTestsClient/RunGrpcTests.ps1 [hunk 2]
```diff
@@ -3,11 +3,15 @@
...
     [bool]$use_http3 = $false,
-    [string]$framework = "net6.0",
+    [bool]$publish_aot = $false,
+    [string]$framework = "net7.0",
     [string]$grpc_web_mode = "None",
     [int]$server_port = 50052
 )
 
+# Command line example:
+# .\RunGrpcTestsAot.ps1 -publish_aot $true
```
Comment (BrennanConroy) [inline testassets/InteropTestsClient/RunGrpcTests.ps1:13]:
```suggestion
# .\RunGrpcTests.ps1 -publish_aot $true
```
Unless you meant to push another ps1 file but missed it

File: testassets/InteropTestsClient/RunGrpcTests.ps1 [hunk 3]
```diff
@@ -3,11 +3,15 @@
     [bool]$use_tls = $false,
     [bool]$use_winhttp = $false,
     [bool]$use_http3 = $false,
-    [string]$framework = "net6.0",
+    [bool]$publish_aot = $false,
```
Comment (BrennanConroy) [inline testassets/InteropTestsClient/RunGrpcTests.ps1:6]: Are you planning on adding a PR test run with this set to true?

File: testassets/InteropTestsClient/InteropTestsClient.csproj [hunk 4]
```diff
@@ -44,8 +40,15 @@
...
   </ItemGroup>
 
-  <ItemGroup Condition="'$(TargetFramework)' == 'net5.0' OR '$(TargetFramework)' == 'net6.0' OR '$(TargetFramework)' == 'net7.0'">
+  <ItemGroup Condition="'$(PublishAot)'!='true'">
+    <ProjectReference Include="..\..\src\Grpc.Auth\Grpc.Auth.csproj" />
+    <ProjectReference Include="..\..\src\Grpc.Net.Client\Grpc.Net.Client.csproj" />
     <ProjectReference Include="..\..\src\Grpc.Net.Client.Web\Grpc.Net.Client.Web.csproj" />
   </ItemGroup>
+  <ItemGroup Condition="'$(PublishAot)'=='true'">
+    <PackageReference Include="Grpc.Auth" Version="$(GrpcDotNetPackageVersion)" />
```
Comment (BrennanConroy) [inline testassets/InteropTestsClient/InteropTestsClient.csproj:49]: What's the reason for this?

File: build/dependencies.props (hunk 1 above)
Comment (JamesNK) [reply build/dependencies.props:19]: I want to leave it as this for now, but I think soon we'll remove old targets from grpc-dotnet and upgrade this package version to something more modern.

File: testassets/InteropTestsClient/InteropTestsClient.csproj (hunk 4 above)
Comment (JamesNK) [reply testassets/InteropTestsClient/InteropTestsClient.csproj:49]: This bug - https://github.com/dotnet/sdk/issues/28169

File: testassets/InteropTestsClient/RunGrpcTests.ps1 (hunk 3 above)
Comment (JamesNK) [reply testassets/InteropTestsClient/RunGrpcTests.ps1:6]: The interop tests are run on google infrastructure. I will organize running them twice, with and without AOT, soon.

**Review by BrennanConroy — APPROVED** [review]

**Pull request merged**